awsmp/models.py

```diff
@@ -139,3 +139,111 @@ def ami_access_role_arn_validator(cls, value):
         if not value.startswith("arn:aws:iam::"):
             raise ValueError(f"{value} is invalid role format. Please check your role again.")
         return value
+
+
+class DescriptionModel(BaseModel):
+    ProductTitle: str
+    ProductCode: str
+    ShortDescription: str
+    LongDescription: str
+    Sku: str
+    Highlights: List[str]
+    SearchKeywords: List[str]
+    Categories: List[str]
+
+class PromotionalResourcesModel(BaseModel):
+    LogoUrl: str
+    Videos: List[str]
+    AdditionalResources: List[dict[str, str]]
+
+class SupportInformationModel(BaseModel):
+    Description: str
+    Resources: List[str]
+
+class RegionAvailabilityModel(BaseModel):
+    Regions: List[str]
+    FutureRegionSupport: str
+
+class Entity(BaseModel):
+    Description: DescriptionModel
+    PromotionalResources: PromotionalResourcesModel
+    SupportInformation: SupportInformationModel
+    RegionAvailability: RegionAvailabilityModel
+
+    @staticmethod
+    def _get_entity(response: dict[str, Any]):
+        return Entity(**response)
+
+    def _get_description_diff(self, local_config: AmiProduct):
+
+        desc = self.Description
+        promo = self.PromotionalResources
+        support = self.SupportInformation
+
+        json_yaml_description_mapping = {
+            "product_title": desc.ProductTitle,
+            "short_description": desc.ShortDescription,
+            "long_description": desc.LongDescription,
+            "highlights": desc.Highlights,
+            "logourl": promo.LogoUrl,
+            "search_keywords": desc.SearchKeywords,
+            "support_description": support.Description,
+            "categories": desc.Categories,
+        }
+
+        live_listing_config = AmiProduct(**json_yaml_description_mapping)
+        diffs = {}
+
+        for k, v in live_listing_config.dict().items():
```

Comment:
this looks like repetition. please extract that code into a general diff functio